File: app/src/main/java/com/timotiusoktorio/pencake/ui/orderdetail/OrderDetailViewModel.java
package com.timotiusoktorio.pencake.ui.orderdetail;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.timotiusoktorio.pencake.data.model.StoreInfo;
import com.timotiusoktorio.pencake.data.source.DataManager;

class OrderDetailViewModel extends ViewModel {

    final MutableLiveData<StoreInfo> storeInfoLiveData = new MutableLiveData<>();

    private final DataManager dataManager;

    OrderDetailViewModel(DataManager dataManager) {
        this.dataManager = dataManager;
        loadStoreInfo();
    }

    private void loadStoreInfo() {
        dataManager.fetchStoreInfo(new DataManager.SingleCallback<StoreInfo>() {
            @Override
            public void onSuccess(StoreInfo data) {
                storeInfoLiveData.setValue(data);
            }

            @Override
            public void onError(String errorMsg) {

            }
        });
    }
}

File: app/src/main/java/com/timotiusoktorio/pencake/ui/orderdetail/OrderDetailViewModel.kt
package com.timotiusoktorio.pencake.ui.orderdetail

import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel
import com.timotiusoktorio.pencake.data.model.StoreInfo
import com.timotiusoktorio.pencake.data.source.DataManager

class OrderDetailViewModel(private val dataManager: DataManager) : ViewModel() {

    val storeInfoLiveData = MutableLiveData<StoreInfo>()

    init {
        loadStoreInfo()
    }

    private fun loadStoreInfo() {
        dataManager.fetchStoreInfo(object : DataManager.SingleCallback<StoreInfo> {
            override fun onSuccess(data: StoreInfo) {
                storeInfoLiveData.value = data
            }

            override fun onError(errorMsg: String) {
                // Not implemented
            }
        })
    }
}